### PR TITLE
Fa 988 add client request

### PR DESCRIPTION
**Pull request by Olson3R**
# Title
[FA-988] add client request
## What?
Adding the ability to get a token based off of a CloudVault application client id and client secret.

If you run your application with a `SENTERA_ENV` environment variable, it will use that to determine the environment. Otherwise it currently defaults to production since we don't have a great setup for external customers to use anything else.
`SENTERA_ENV=development pipenv run python myalert.py`

## Why?
It can be used by application processes.

## QA Strategy
- [x] Merged latest master
- [x] Updated version number
- [x] Can get an access token based on client credentials
- [x] Can get an access token based on user credentials
- [x] Can create an alert
- [x] Can query for weather data

## Breaking Changes
- changed `dtn_key` to `sentera_api_key`

### CODE REVIEW

File: sentera/weather.py [hunk 1]
```diff
@@ -284,7 +289,7 @@ def _merge_to_full_df(weather_variable, weather_interval, response_json, data_df
     time_interval_list,
     weather_interval,
     weather_type,
-    dtn_key=None,
+    sentera_api_key=None,
```
Comment (Olson3R) [inline sentera/weather.py]: It isn't specific to DTN.

File: sentera/configuration.ini [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+sentera_api_url = https://apistaging.sentera.com
+weather_api_url = https://weatherstaging.sentera.com
+
+[staging2]
+sentera_api_url = https://apistaging2.sentera.com
+weather_api_url = https://weatherstaging2.sentera.com
+
+[test]
+sentera_api_url = https://apitest.sentera.com
+weather_api_url = https://weathertest.sentera.com
```
Comment (Olson3R) [inline sentera/configuration.ini]: These are purposely urls that do not work.

File: sentera/configuration.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+"""Evnironment specific configurations."""
```
Comment (DanSchoppe) [inline sentera/configuration.py]:
```suggestion
"""Environment specific configurations."""
```

File: sentera/configuration.ini (hunk 2 above)
Comment (DanSchoppe) [reply sentera/configuration.ini]: Maybe mention that as a comment here?

File: sentera/configuration.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
+
+
+class Configuration:
+    """Class toretrieve configurations from."""
```
Comment (DanSchoppe) [inline sentera/configuration.py]:
```suggestion
    """Class to retrieve configurations from."""
```

File: sentera/configuration.py (hunk 3 above)
Comment (ZachMarston) [reply sentera/configuration.py]: Good catch

File: sentera/configuration.ini (hunk 2 above)
Comment (Olson3R) [reply sentera/configuration.ini]: Done.

File: sentera/weather.py [hunk 5]
```diff
@@ -124,7 +125,9 @@ def build_weather_url(weather_type, weather_variable, weather_interval, lat, lon
     :return: Constructed query URL, as string.
     """
     if weather_type == WeatherType.SevenDay:
-        return posixpath.join(WEATHER_BASE_URL, str(weather_type), str(lat), str(long),)
+        return configuration.weather_api_url(
+            "/{}/{}/{}".format(str(weather_type), str(lat), str(long))
+        )
```
Comment (DanSchoppe) [inline sentera/weather.py]: Curious why you dropped `path.join()` operations in favor of string interpolation? Seems like path.join() would be preferable here.

**Comment by DanSchoppe** [comment]
Looks awesome! 🎉

File: sentera/weather.py (hunk 5 above)
Comment (Olson3R) [reply sentera/weather.py]: I prefer string interpolation I guess.

Comment (Olson3R) [reply sentera/weather.py]: The one below needed to use interpolation for one of the arguments anyhow.

Comment (DanSchoppe) [reply sentera/weather.py]: Ok, thanks!

File: sentera/tests/test_auth.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    httpretty.register_uri(
+        httpretty.POST, "https://apitest.sentera.com/oauth/token", body=request_callback
+    )
+    response = get_application_token("test_client_id_123", "test_client_secret_321")
+    assert response["access_token"] == "my-test-access-token-abc123"
+    assert len(httpretty.latest_requests()) == 1
+
+
+@httpretty.httprettified
+def test_get_auth_token_success():
```
Comment (jdalt) [inline sentera/tests/test_auth.py]: Is this use of the sessions endpoint for backwards compatibility?

Comment (Olson3R) [reply sentera/tests/test_auth.py]: Sort of. It is possibly someone wants to make a CLI application that uses their username and password I suppose. I feel like that would still be a valid use.

Comment (jdalt) [reply sentera/tests/test_auth.py]: I'd like that to go through oauth similar to the firebase flow, but that's a fairly heavy lift...so until we retire `/v1/sessions` I guess I'm cool with it.

File: sentera/weather.py (hunk 1 above)
Comment (elliott-imhoff) [reply sentera/weather.py]: This key actually is specific to dtn.  We have two different keys floating around right now, one of which is used to access dtn data, and the other is for sentera data.

Comment (elliott-imhoff) [reply sentera/weather.py]: nvm!

File: Pipfile [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+pytest = "*"
+httpretty = "*"
+pre-commit = "*"
+
+[packages]
+requests = "*"
+pandas = "*"
+aiohttp = "*"
+tqdm = "*"
+
```
Comment (elliott-imhoff) [inline Pipfile]: Do you not have to specify ``sentera`` as a package?

**Comment by elliott-imhoff** [comment]
https://github.com/SenteraLLC/py-sentera-api/blob/41a62730b074e1596a7e8a9ed27b32d392b6e689/sentera/api.py#L58

dtn_key is still used here

File: Pipfile (hunk 7 above)
Comment (Olson3R) [reply Pipfile]: No, it isn't an external package. The code _is_ the repo.

File: sentera/tests/test_auth.py (hunk 6 above)
Comment (Olson3R) [reply sentera/tests/test_auth.py]: Yeah, I expect we will do more work with the user flow.

**Pull request merged**